coderd/database/dbmem/dbmem.go

```diff
@@ -6493,6 +6534,36 @@ func (q *FakeQuerier) InsertProvisionerJobLogs(_ context.Context, arg database.I
 	return logs, nil
 }
 
+func (q *FakeQuerier) InsertProvisionerKey(_ context.Context, arg database.InsertProvisionerKeyParams) (database.ProvisionerKey, error) {
+	err := validateDatabaseType(arg)
+	if err != nil {
+		return database.ProvisionerKey{}, err
+	}
+
+	q.mutex.Lock()
+	defer q.mutex.Unlock()
+
+	newErr := *errUniqueConstraint
+	newErr.Constraint = string(database.UniqueProvisionerKeysOrganizationIDNameKey)
```

Comment:
Feels like this should be some helper function to prevent accidental modification of the original error pointer. I know this is safe, but feels like something that could be broken by a refactor or something.